larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewTransverseTracksAlgorithm.cc

```diff
@@ -101,13 +100,16 @@ pandora::StatusCode TwoViewTransverseTracksAlgorithm::CalculateOverlapResult(con
     const float correlation(LArDiscreteProbabilityHelper::CalculateCorrelationCoefficient(
         resampledDiscreteProbabilityVector1, resampledDiscreteProbabilityVector2));
 
+    std::mt19937 randomNumberGenerator(static_cast<std::mt19937::result_type>(
```

Comment:
I did ponder whether to keep the RNG as a member when making this change but opted to drop it.  Because, I could imagine a situation where a future edit to the code structure could result in a logic hole where the seed doesn't get changed for some events.   This would mean that the same seed is used for some adjacent events, erroneously.

In terms of optimisation I can certainly see why it would be better to initialise the RNG once and repeatedly change the state rather than repeatedly initialise it.

Maybe I'm being too superstitious, what do you think @AndyChappell ?

Reply:
Presumably you could choose to set the seed for the RNG earlier in the code, in particular the top-level CalculateOverlapResult function? Given that this function always gets called as part of the template for the MatchingBaseAlgorithm I think it's fairly safe to assume that this would be insulated from structural changes. Also, I just noticed this code uses the size of pCluster2 twice for the seed, presumably this should have been pCluster1 and pCluster2?

Reply:
Sounds like a good solution to me.
I've made the requested change.
I also fixed the benign seed setting bug; absolutely it should have been the sum of p1 and p2 hits.  I don't think it really matters in terms of output as p2+p2 is still a valid seed, and the output shouldn't depend too much on the seed, but it was certainly not intended.  Good catch!